File: backend/components/metiers/administration_publique/tests/utils/i18n/i18n_utils.test.py
# i18n_utils.test.py – Tests unitaires ultra avancés pour i18n_utils.py
import pytest
from .i18n_utils import translate, is_supported_lang, SUPPORTED_LANGS

def test_translate_supported_langs():
    for lang in SUPPORTED_LANGS:
        assert f'[{lang.upper()}]' in translate('Test', lang)

def test_translate_unsupported_lang():
    assert translate('Test', 'xx') == 'Test'

def test_is_supported_lang():
    assert is_supported_lang('fr')
    assert is_supported_lang('en')
    assert not is_supported_lang('xx')

File: backend/components/metiers/administration_publique/tests/utils/i18n/i18n_utils.test.js
// i18n_utils.test.js – Tests unitaires ultra avancés pour i18n_utils.js
const i18n = require('./i18n_utils');

describe('i18n_utils – Traduction', () => {
  it('traduit dans toutes les langues supportées', () => {
    i18n.SUPPORTED_LANGS.forEach(lang => {
      expect(i18n.translate('Test', lang)).toMatch(new RegExp(`\[${lang.toUpperCase()}\]`));
    });
  });
  it('retourne le message brut si langue non supportée', () => {
    expect(i18n.translate('Test', 'xx')).toBe('Test');
  });
});

describe('i18n_utils – Langues supportées', () => {
  it('reconnaît les langues supportées', () => {
    expect(i18n.isSupportedLang('fr')).toBe(true);
    expect(i18n.isSupportedLang('en')).toBe(true);
    expect(i18n.isSupportedLang('xx')).toBe(false);
  });
});
